numba/core/opt_info.py

```diff
@@ -0,0 +1,183 @@
+from typing import Any, Iterable, List, Set, Tuple
+from collections import namedtuple
+
+
+class OptimizationProcessor:
+    """
+    A converter that reads LLVM optimization remarks and produced some useful
+    information for the programmer
+    """
+    def filters(self) -> Iterable[str]:
+        """
+        Provide a list of identifiers that will be requested during compilation.
+        These can be regular expressions.
+        """
+        pass
+
+    def needs_debug_info(self) -> bool:
+        """
+        Determines if the processor requires debugging information (file and
+        line numbers) or not.
+
+        The default is to require it, but, if overridden to false, this can
+        save time at compilation.
+        """
+        return True
+
+    def process(self, remarks_data: List[Any], full_name: str) ->\
+            Iterable[Tuple[str, Any]]:
+        """
+        Generate optimisation information from the remarks.
+
+        Parameters
+        ----------
+        remarks_data : a list of the parsed YAML provided by LLVM; there may
+        be multiple compilation passes and remarks will be generated for
+        each pass. Note that this data is shared and must not be mutated.
+        full_name : the complete Numba name for the function being compiled
+
+        Returns
+        -------
+        A collection of key-value pairs with the key as a unique name and the
+        value as the optimisation information.
+        """
+        pass
+
+
+class Aggregate(OptimizationProcessor):
+    """
+    A processor which is a collection of other processors
+    """
+    def __init__(self, *processors):
+        self._processors = processors
+
+    def filters(self) -> Iterable[str]:
+        return (f for p in self._processors for f in p.filters())
+
+    def needs_debug_info(self) -> bool:
+        return any(p.needs_debug_info() for p in self._processors)
+
+    def process(self, remarks_data: List[Any], full_name: str) ->\
+            Iterable[Tuple[str, Any]]:
+        return (o for p in self._processors for o in p.process(remarks_data,
+                                                               full_name))
+
+
+class RawOptimizationRemarks(OptimizationProcessor):
+    """
+    This stores raw optimisation remarks
+
+    It is almost certainly a bad idea to use as the remarks are large and
+    inefficient. This is intended as a utility for viewing remarks
+    information when developing new processors.
+    """
+
+    def __init__(self, *filter_names):
+        self.filter_names = set(filter_names)
+
+    def filters(self) -> Iterable[str]:
+        return self.filter_names
+
+    def needs_debug_info(self) -> bool:
+        return False
+
+    def process(self, remarks_data: List[Any], function) -> Iterable[
+            Tuple[str, Any]]:
+        return ('raw', remarks_data),
+
+
+class LoopDeLoop(OptimizationProcessor):
```

Comment:
Why not just name this `LoopVectorizationDetector` since it is entirely focusing on loop vectorization?